src/kagglehub/models.py

```diff
@@ -18,5 +25,29 @@ def model_download(handle: str, path: Optional[str] = None):
     return registry.resolver(h, path)
 
 
-def model_upload():
-    raise NotImplementedError()
+def model_upload(
+    handle: str, local_model_dir: str, license_name: str, model_type: str, version_notes: Optional[str] = None
+):
+    """Upload model files.
+
+    Args:
+        handle: (string) the model handle.
+        local_model_dir: (string) path to a file in a local directory.
+        license_name: (string) model license.
+        version_notes: (string) Optional to write to model versions.
+    """
+    # parse slug
+    h = parse_model_handle(handle)
+
+    if h.is_versioned():
+        is_versioned_exception = "The model handle should not include the version"
+        raise ValueError(is_versioned_exception)
+
+    # Create the model if it doesn't already exist
+    get_or_create_model(h.owner, h.model)
```

Comment:
nit: you are not really "getting" the model here. Maybe rename to `ensure_model` or `create_model_if_missing`.